nemo_skills/pipeline/utils.py

```diff
@@ -131,6 +143,82 @@ def check_if_mounted(cluster_config, path_to_check):
         raise ValueError(f"The path '{path_to_check}' is not mounted. Check cluster config.")
 
 
+def check_remote_mounts(
+    cluster_config, log_dir: str, mount_map: Dict[str, Optional[str]] = None, check_mounted_paths: bool = False
+):
+    """
+    Utility method to check if the paths are mounted, whether the remote directories exist, create them if they dont exist
+    and finally resolve their mounted paths.
+
+    Args:
+        cluster_config: cluster config dictionary
+        log_dir: optional str representing the log directory
+        mount_map: a dictionary mapping the paths to their default mount locations.
+            Keys must be the mount source, and values must be the mount destination.
+            If the mount destination is None, the path will not be mounted but still checked and remote directory will still be created.
+            Mount destinations must be absolute paths and begin with '/'.
+        check_mounted_paths: if True, will perform remote calls to dynamically mount the provided paths, and create
+            remote directories if required. Finally, will assert that all remote paths are valid files or directories.
+
+    Returns:
+        tuple: a tuple of the mounted paths for the provided paths in mount_map, and the log_dir
+    """
+    # Check if mount map is provided
+    mount_map = mount_map or {}
+
+    # Compute directory of all files
+    remote_dir_list = []
+
+    # Check paths and add to mount list if not mounted
+    if check_mounted_paths:
+        for idx, (path, default_mount) in enumerate(mount_map.items()):
+            if not is_mounted_filepath(cluster_config, path):
+                # check if the path is a file or a directory
+                # so that the directory can be created
+                is_file = os.path.splitext(path) != ""
```

Comment:
this will fail on any files without extension? Can also send a proper check command through the ssh tunnel, but if you think this is enough, I guess it's ok. But can lead to silent issues later on

Reply:
Its meant to be a dir check - if is a file a filepath with extension or a dir path with no extension. I tested it in python 3.10 - works fine.

Reply:
right, but this will treat any files without extension as a directory and try to create it. Then it's unclear how the downstream code will interact with that and might lead to silent errors. We don't normally use file names without extensions, but it's certainly possible to do that, right?

Reply:
Yes possible to do that.. Hmm this part is actually only necessary for 2 scripts that try to create output directory. If i manually do it in those two scripts, then i can skip this logic here. Let me change that in another pr.

Reply:
Made it just a check and raise error in another PR.